src/main/kotlin/app/revanced/patches/youtube/layout/esportmode/patch/EsportModePatch.kt

```diff
@@ -0,0 +1,72 @@
+package app.revanced.patches.youtube.layout.esportmode.patch
+
+import app.revanced.patcher.annotation.Description
+import app.revanced.patcher.annotation.Name
+import app.revanced.patcher.annotation.Version
+import app.revanced.patcher.data.impl.BytecodeData
+import app.revanced.patcher.extensions.addInstructions
+import app.revanced.patcher.patch.PatchResult
+import app.revanced.patcher.patch.PatchResultSuccess
+import app.revanced.patcher.patch.annotations.DependsOn
+import app.revanced.patcher.patch.annotations.Patch
+import app.revanced.patcher.patch.impl.BytecodePatch
+import app.revanced.patches.youtube.layout.esportmode.annotations.EsportModeCompatibility
+import app.revanced.patches.youtube.layout.esportmode.fingerprints.InlineTimeBarWrapperFingerprint
+import app.revanced.patches.youtube.layout.esportmode.fingerprints.TimeCounterFingerprint
+import app.revanced.patches.youtube.misc.integrations.patch.IntegrationsPatch
+import app.revanced.patches.youtube.misc.settings.bytecode.patch.SettingsPatch
+import app.revanced.patches.youtube.misc.settings.framework.components.impl.StringResource
+import app.revanced.patches.youtube.misc.settings.framework.components.impl.SwitchPreference
+
+@Patch
+@DependsOn([IntegrationsPatch::class, SettingsPatch::class])
+@Name("e-sport-mode")
+@Description("Hides progress bar and time counter on videos.")
+@EsportModeCompatibility
+@Version("0.0.1")
+class EsportModePatch : BytecodePatch(
+    listOf(
+        InlineTimeBarWrapperFingerprint, TimeCounterFingerprint
+    )
+) {
+    override fun execute(data: BytecodeData): PatchResult {
+        SettingsPatch.PreferenceScreen.LAYOUT.addPreferences(
+            SwitchPreference(
+                "revanced_esport_mode_enabled",
+                StringResource("revanced_esport_mode_title", "Enable e-Sport Mode"),
+                false,
+                StringResource("revanced_esport_mode_summary_on", "e-Sport Mode is enabled"),
+                StringResource("revanced_esport_mode_summary_off", "e-Sport Mode is disabled")
+            )
+        )
+
+        val inlineTimeBarWrapperMethod = InlineTimeBarWrapperFingerprint.result!!.mutableMethod
+
+        inlineTimeBarWrapperMethod.addInstructions(
+            0, """
+            invoke-static { }, Lapp/revanced/integrations/patches/EsportModePatch;->esportModeEnabled()Z
+            move-result v0
+            if-eqz v0, :esport_mode_enabled
+            return-void
+            :esport_mode_enabled
+            nop
+        """
+        )
+
+        val timeCounterMethod = TimeCounterFingerprint.result!!.mutableMethod
+
+        timeCounterMethod.addInstructions(
+            0, """
+            invoke-static { }, Lapp/revanced/integrations/patches/EsportModePatch;->esportModeEnabled()Z
+            move-result v0
+            if-eqz v0, :esport_mode_enabled
+            const-string v0, "xx:xx:xx"
```

Comment:
```suggestion
            const-string v0, ""
```

Reply:
A user explictly request me to add this as time counter, instead nothing.

Reply:
Other users will request to remove. if you want to keep it, make it toggleable on-click.